diff --git a/node_modules/xpath/xpath.d.ts b/node_modules/xpath/xpath.d.ts
index 8967885..1c47f44 100644
--- a/node_modules/xpath/xpath.d.ts
+++ b/node_modules/xpath/xpath.d.ts
@@ -2,10 +2,12 @@
 
 type SelectedValue = Node | Attr | string | number | boolean;
 interface XPathSelect {
-    (expression: string, node?: Node): Array<SelectedValue>;
+    (expression: string, node?: Node): Array<SelectedValue> | SelectedValue;
     (expression: string, node: Node, single: true): SelectedValue | undefined;
 }
 export var select: XPathSelect;
 export function select1(expression: string, node?: Node): SelectedValue | undefined;
 export function evaluate(expression: string, contextNode: Node, resolver: XPathNSResolver | null, type: number, result: XPathResult | null): XPathResult;
 export function useNamespaces(namespaceMap: { [name: string]: string }): XPathSelect;
+export function parse(expression: string): XPathEvaluator
+type XPathEvaluator = Object;
